src/geom/SkyWcs.cc

```diff
@@ -44,44 +50,114 @@ namespace afw {
 namespace geom {
 namespace {
 
+int const SERIALIZATION_VERSION = 1;
+
+double const TIGHT_FITS_TOL = 0.0001;
+
 inline double square(double x) { return x * x; }
 
+class SkyWcsPersistenceHelper {
+public:
+    table::Schema schema;
+    table::Key<table::Array<std::uint8_t>> wcs;
+
+    static SkyWcsPersistenceHelper const& get() {
+        static SkyWcsPersistenceHelper instance;
+        return instance;
+    }
+
+    // No copying
+    SkyWcsPersistenceHelper(const SkyWcsPersistenceHelper&) = delete;
+    SkyWcsPersistenceHelper& operator=(const SkyWcsPersistenceHelper&) = delete;
+
+    // No moving
+    SkyWcsPersistenceHelper(SkyWcsPersistenceHelper&&) = delete;
+    SkyWcsPersistenceHelper& operator=(SkyWcsPersistenceHelper&&) = delete;
+
+private:
+    SkyWcsPersistenceHelper()
+            : schema(),
+              wcs(schema.addField<table::Array<std::uint8_t>>("wcs", "wcs string representation", "")) {
+        schema.getCitizen().markPersistent();
+    }
+};
+
+class SkyWcsFactory : public table::io::PersistableFactory {
+public:
+    explicit SkyWcsFactory(std::string const& name) : table::io::PersistableFactory(name) {}
+
+    virtual std::shared_ptr<table::io::Persistable> read(InputArchive const& archive,
+                                                         CatalogVector const& catalogs) const {
+        SkyWcsPersistenceHelper const& keys = SkyWcsPersistenceHelper::get();
+        LSST_ARCHIVE_ASSERT(catalogs.size() == 1u);
+        LSST_ARCHIVE_ASSERT(catalogs.front().size() == 1u);
+        LSST_ARCHIVE_ASSERT(catalogs.front().getSchema() == keys.schema);
+        table::BaseRecord const& record = catalogs.front().front();
+        std::string stringRep = formatters::bytesToString(record.get(keys.wcs));
```

Comment:
It's because we're connecting one persistence framework (`table::io`) to another (AST's) by dumping the AST-serialized byte stream into a single variable-length uint8 array `afw::table` field.